### PR TITLE
pre-commit hook, readme updates

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
## Summary

Checks-off one part of https://github.com/wnfs-wg/rs-wnfs/issues/71.

Standardizes formatting, linting, etc, on pre-commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,3 +257,36 @@ async fn main() {
...
+For usage questions, usecases, or issues reach out to us in our [Discord webnative-fs channel](https://discord.com/channels/478735028319158273/877623827331428403).
+We would be happy to try to answer your question or try opening a new issue on Github.
+
+## External Resources
+
+This is a list of links to blog posts, confereence talks, and tutorials related to WNFS in general:
+
+### Talks
+
+- [A Distributed File System for Secure P2P Applications](https://www.youtube.com/watch?v=-f4cH_HQU4U) by Brooklyn Zelenka (Strange Loop 2022)
```
Comment (matheus23) [inline README.md:282]: There's also a short overview of WNFS from IPFS Thing: https://www.youtube.com/watch?v=3se17NAS-Lw

**Review by matheus23 — APPROVED** [review]
Thumbs up, you got it

Sweet! Git pre-commit hooks will be great. I can't count the times I've pushed something without testing :sweat_smile: (even just today)

File: README.md (hunk 1 above)
Comment (zeeshanlakhani) [reply README.md:282]: good call, will add.

**Comment by zeeshanlakhani** [comment]
Will fix the thresholding :)

**Comment by zeeshanlakhani** [comment]
@matheus23 we don't run tests on the pre-commit, cause that can take time if not incrementally-cache compiled, but good for clippy, formatting, and other such things.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+repos:
+  - repo: local
+    hooks:
+      - id: fmt
+        name: fmt
+        description: Format rust files.
+        entry: cargo +nightly fmt
+        language: system
+        types: [rust]
+        args: ["--all", "--"]
```
Comment (appcypher) [inline .pre-commit-config.yaml:11]: Adding the `--check` flag shows a more detailed error message.
```suggestion
        args: ["--all", "--", "--check"]
```

**Pull request merged**